src/zenml/analytics/context.py

```diff
@@ -78,9 +78,19 @@ def __enter__(self) -> "AnalyticsContext":
 
         try:
             gc = GlobalConfiguration()
+
+            if not gc.is_initialized:
+                # If the global configuration is not initialized, using the
+                # zen store can lead to multiple initialization issues, because
+                # the analytics are triggered during the initialization of the
+                # zen store.
+                return self
+
             store_info = gc.zen_store.get_store_info()
 
-            if self.in_server:
+            # For local ZenML servers, we always use the client's analytics
+            # opt-in configuration.
+            if self.in_server and store_info.is_local():
```

Comment:
@stefannica I think this is still not correct: Shouldn't it be `and not store_info.is_local()`?

In any case, I think `is_local` does not work for the in-server case: When we start a local server, we create a new global configuration for this server (in `<GLOBAL_CONFIG_DIR>/zen_server/daemon/.zenconfig`). In this global config file:
- the `user_id` which is used for the `is_local()` check is not the same as the `user_id` stored in the sqlite databast, which means the check probably returns `False`
- What's even more alarming, this global config file has `analytics_opt_in=true`? Do we somehow overwrite that value with an environment variable? I see some code for that for the docker one, but not the `DaemonZenServer`. Potentially this get's picked up from the local env, but this all seems to be quite a convoluted mess 😅

Reply:
Nice, this is much better now that we use the same `client_id` in both global configs.
This means that the analytics events for local servers will now have the same client ID in segment events, but that should not be a problem as far as I know. @htahir1 what do you think?

Reply:
I talked with @htahir1 and this should be fine